### ISSUE TITLE
Can't quit the app when it's minimized to the system tray

### ISSUE THREAD

**Issue by guihkx** [nuttyartist/notes#461]
This only affects the Qt 6 build for some reason, but I could reproduce the problem on both Linux and Windows:

1. Close Notes to system tray
2. Right click the system tray icon
3. Choose 'Quit'

**Expected result:**

The app should quit, like the Qt 5 build does.

**Current result:**

The app does not quit; you have to restore it from the system tray and then quit.

**Comment by nuttyartist** [comment]
Ugh, that's annoying. If we can't progress on this ourselves, we should probably file an issue to Qt?

**Comment by guihkx** [comment]
It might be a Qt bug, indeed. Especially since it worked on Qt 5.

But I was able to work around it by replacing `MainWindow::close()` by `qApp->quit()` here:

https://github.com/nuttyartist/notes/blob/1142bce844839c6757b2c94f3c7ffa6a92494844/src/mainwindow.cpp#L2189-L2193

And it works on both Qt 5 and Qt 6.

Is that wrong? Judging by the output I got from [`AddressSanitizer`](https://github.com/google/sanitizers/wiki/AddressSanitizer) upon exiting Notes, there hasn't been any additional memory leaks (apart from the ones we already had 😅), so I'm gonna go with 'no'.

I made sure to check the AddressSanitizer output on both Qt 5 and Qt 6 builds, by the way.

But I'll open a PR for reviews, just in case. :)
